File: TypeScript/0088-merge-sorted-array.ts
// 2020-08-08 15:50:07

/**
 Do not return anything, modify nums1 in-place instead.
 */
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
  let k = nums1.length - 1;
  let i = m - 1;
  let j = n - 1;
  while (i >= 0 && j >= 0) {
    if (nums1[i] < nums2[j]) {
      nums1[k--] = nums2[j--];
    } else {
      nums1[k--] = nums1[i--];
    }
  }

  while (j >= 0) {
    nums1[k--] = nums2[j--]
  }
};

File: C++/0088_mergesortedarray.cc
// 2020-08-08 23:57:39

#include <vector>

using std::vector;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int total = nums1.size() - 1;
        int i = m - 1;
        int j = n - 1;

        while (i >= 0 && j >= 0) {
            if (nums1[i] < nums2[j]) {
                nums1[total--] = nums2[j--];
            } else {
                nums1[total--] = nums1[i--];
            }
        }

        while (j >= 0) {
            nums1[total--] = nums2[j--];
        }

    }
};